### ISSUE TITLE
Variables inside array arguments aren't processed correctly

### ISSUE THREAD

**Issue by kipz** [walmartlabs/lacinia#286]
The following query causes an exception to be thrown:

```graphql
query ($integerArray1: Int, $integerArray2: Int) {
    echoArgs(integerArray: [$integerArray1, $integerArray2, 10, 100]){
       integerArray
    }
}
```

```
clojure.lang.ExceptionInfo: Exception applying arguments to field `echoArgs': For argument `integerArray', no method in multimethod 'process-dynamic-argument' for dispatch value: :array
 at clojure.core$ex_info.invokeStatic (core.clj:4739)
    clojure.core$ex_info.invoke (core.clj:4739)
    com.walmartlabs.lacinia.internal_utils$throw_exception.invokeStatic (internal_utils.clj:156)
    com.walmartlabs.lacinia.internal_utils$throw_exception.invoke (internal_utils.clj:149)
    com.walmartlabs.lacinia.parser$eval4272$fn__4273$fn__4277$fn__4281.invoke (parser.clj:1035)
    com.walmartlabs.lacinia.parser$eval4272$fn__4273$fn__4277.invoke (parser.clj:1030)
    com.walmartlabs.lacinia.parser$eval4272$fn__4273.invoke (parser.clj:1018)
    clojure.lang.MultiFn.invoke (MultiFn.java:238)
    com.walmartlabs.lacinia.parser$xform_query$fn__4316.invoke (parser.clj:1185)
    clojure.core$mapv$fn__8088.invoke (core.clj:6832)
    clojure.lang.PersistentVector.reduce (PersistentVector.java:341)
    clojure.core$reduce.invokeStatic (core.clj:6747)
    clojure.core$mapv.invokeStatic (core.clj:6823)
    clojure.core$mapv.invoke (core.clj:6823)
    com.walmartlabs.lacinia.parser$xform_query.invokeStatic (parser.clj:1185)
    com.walmartlabs.lacinia.parser$xform_query.invoke (parser.clj:1157)
    com.walmartlabs.lacinia.parser$parse_query.invokeStatic (parser.clj:1219)
    com.walmartlabs.lacinia.parser$parse_query.invoke (parser.clj:1206)
    com.walmartlabs.lacinia.parser$parse_query.invokeStatic (parser.clj:1213)
    com.walmartlabs.lacinia.parser$parse_query.invoke (parser.clj:1206)
```
I have a fork with a commit that seems to work for me, but it doesn't look as complete as the `:object` handling, so it might not be ready for a PR:

https://github.com/kipz/lacinia/commit/797b0dec98cc49164bd296215c0ebfb4af93808a

**Comment by hlship** [comment]
I vaguely remember deliberately punting on variables inside lists as it added overhead and complication and I wasn't sure anyone needed it.

**Comment by kipz** [comment]
Yeah, I get that. For us though, it's mostly about making sure queries that work in apollo (which we use for stitching) also works in our clojure endpoints.

**Comment by hlship** [comment]
One of our internal users hit a similar problem with input objects (not lists) but I think it is the same cause and I believe I have a solution.

**Comment by hlship** [comment]
Well, no different causes, but let's see if I can fix this as well.

**Comment by hlship** [comment]
Fixed in 073b5abb31f3fcebf67355bfb959e6a54ac83a43
